src/vmcs/src/ept_intel_x64.cpp

```diff
@@ -41,60 +41,66 @@ ept_intel_x64::ept_intel_x64(pointer epte)
 ept_entry_intel_x64
 ept_intel_x64::add_page(integer_pointer gpa, integer_pointer bits, integer_pointer end)
 {
-    auto &&index = ept::index(gpa, bits);
+    auto index = ept::index(gpa, bits);
+    auto view = gsl::make_span(m_ept, ept::num_entries);
```

Comment:
@rianquinn are you set on making it a constructor in the entry class?  I think it may be better just to make a function in ept_intel_x64 that makes the view.  This way the private m_ept wouldn't need to be passed to the entry class